test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,171 @@
+      assertEquals(path + ":1: Testing with Defined Position\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+
+
+      // print test with NoPosition
+      reporter print(noPos, "test", reporter.INFO)
+      assertEquals("test\n", err_out.toString)
+      err_out.reset
+
+      reporter print(noPos, "test", reporter.WARNING)
+      assertEquals("warning: test\n", err_out.toString)
+      err_out.reset
+
+      reporter print(noPos, "test", reporter.ERROR)
+      assertEquals("error: test\n", err_out.toString)
+      err_out.reset
+
+
+
+      // print test with DefinedPosition
+      reporter print(posWithFile, "test", reporter.INFO)
+      assertEquals(path + ":1: test\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter print(posWithFile, "test", reporter.WARNING)
+      assertEquals(path + ":1: warning: test\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter print(posWithFile, "test", reporter.ERROR)
+      assertEquals(path + ":1: error: test\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+
+
+      // printColumnMarker test
+      reporter printColumnMarker(noPos)
+      assertEquals("", err_out.toString)
+      err_out.reset
+
+      reporter printColumnMarker(posWithFile)
+      assertEquals("    ^\n", err_out.toString)
+      err_out.reset
+
+
+
+      // display test with NoPosition
+      reporter display(noPos, "Testing display", reporter.INFO)
+      assertEquals("Testing display\n", err_out.toString)
+      err_out.reset
+
+      reporter display(noPos, "Testing display", reporter.WARNING)
+      assertEquals("warning: Testing display\n", err_out.toString)
+      err_out.reset
+
+      reporter display(noPos, "Testing display", reporter.ERROR)
+      assertEquals("error: Testing display\n", err_out.toString)
+      err_out.reset
+
+
+
+      // display test with DefinedPosition
+      reporter display(posWithFile, "Testing display", reporter.INFO)
+      assertEquals(path + ":1: Testing display\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter display(posWithFile, "Testing display", reporter.WARNING)
+      assertEquals(path + ":1: warning: Testing display\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter display(posWithFile, "Testing display", reporter.ERROR)
+      assertEquals(path + ":1: error: Testing display\n" + content + "\n    ^\n", err_out.toString)
+      err_out.reset
+
+      reporter.ERROR.count = 101
+      reporter display(posWithFile, "Testing display2", reporter.ERROR)
+      assertEquals("", err_out.toString)
+      err_out.reset
+
+
+
+      // finish test
+      reporter.ERROR.count = 0
+      reporter finish;    // don't know why but I'm requiring ';' here, otherwise the file does not compile
+      assertEquals("", err_out.toString)
+      err_out.reset
+
+      reporter.WARNING.count = 3
+      reporter.ERROR.count = 10
```

Comment:
Probably when the reporter API is nailed down some more, this should be disallowed. Maybe Reporter should have `protected increment(severity)`.